### PR TITLE
Feature/student websocket connection

### LINKED ISSUE #81: Integracja studenta z serwerem przez websockety
Zgodnie z ustalonym standardem w #70

### LINKED ISSUE #127: Zapisywanie storage do local storage
Ogarnąć czy jest jakiś prosty sposób na usunięcie local storage (W trakcie zmian w aplikacji).

### PR DESCRIPTION
**Pull request by ostatni5**
to clear localStorage  use `localStorage.clear()` in console 

close #127 
close #81

### CODE REVIEW

File: server/websockets.ts [hunk 1]
```diff
@@ -30,8 +30,10 @@ const setupWebSocketServer = () => {
 };
 
 function handlerSubscribeStudent(parsed: StudentSubPayload, ws: WebSocketClient) {
-    const selectedLecture: Lecture | undefined = [...lectures.values()].find((lecture) => lecture.link === parsed.data.lecture_link);
+    const selectedLecture: Lecture | undefined = linkLectureMap.get(parsed.data.lecture_link);
     const selectedStudent: Student | undefined = selectedLecture?.studentList.getStudent(parsed.data.student_id);
+    console.log(parsed.data)
```
Comment (Petros9) [inline server/websockets.ts:35]: I reckon that this function is going to be removed in the future as it is used only for testing

File: src/lecturer/components/sendQuizView/SendQuizView.tsx [hunk 2]
```diff
@@ -126,11 +124,11 @@ export function SendQuizView() {
                             {`Upewnij się, że wszystko jest prawidłowo ustawione i wyślij quiz:`}
                         </Typography>
                         <List>
-                        <ListItem>
+                            <ListItem>
                                 <ListItemText
                                     primary="Nazwa przydzielonego quizu:"
                                     secondary={
-                                        (store.quizes[store.selectedQuiz] != undefined) ? store.quizes[store.selectedQuiz].title : ""
+                                        store.quizes[store.selectedQuiz] ?? ""
```
Comment (Petros9) [inline src/lecturer/components/sendQuizView/SendQuizView.tsx:131]: what does double question mark means? ("??")

File: src/lecturer/services/StoreService.tsx [hunk 3]
```diff
@@ -1,12 +1,51 @@
-import { createContext, ReactNode, useState } from "react";
+import { createContext, ReactNode, useEffect, useState } from "react";
 
 export interface StoreProps {
     children: ReactNode
 }
 
+enum StorageKey {
```
Comment (Petros9) [inline src/lecturer/services/StoreService.tsx:7]: why did you decide to use enum here?

File: src/student/services/StoreService.tsx [hunk 4]
```diff
@@ -1,40 +1,103 @@
...
-    const [quizes, setQuizes] = useState<Quiz[]>([]);
+enum StorageKey {
+    invitation,
+    studentNick,
+    studentId,
+    quizes
+}
+
+const stringKey = (key: StorageKey) => {
+    return "lazare.student." + key;
```
Comment (Petros9) [inline src/student/services/StoreService.tsx:22]: why do you use "lazare..." annotation? is it only for easter-egg-like features or it does have a deeper meaning?

**Review by Petros9 — APPROVED** [review]
Overall I think that you have done a great job (as always). I do really like your diligence and professionalism which are visible in your code.

File: src/lecturer/services/StoreService.tsx (hunk 3 above)
Comment (ostatni5) [reply src/lecturer/services/StoreService.tsx:7]: To get autocompletion from typescript (xD) and to be sure that a don't repeat any key.

File: src/lecturer/components/sendQuizView/SendQuizView.tsx (hunk 2 above)
Comment (ostatni5) [reply src/lecturer/components/sendQuizView/SendQuizView.tsx:131]: it is "shorter if" with fall back value if tested is "null" or "undefined"

https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

File: src/student/services/StoreService.tsx (hunk 4 above)
Comment (ostatni5) [reply src/student/services/StoreService.tsx:22]: Just convention to name keys in storage.
It was my personal choice(not some standard etc), but I think it looks cool xD

File: server/websockets.ts (hunk 1 above)
Comment (ostatni5) [reply server/websockets.ts:35]: Good point it is unnecessary now.

**Review by rozchlastywacz — APPROVED** [review]

**Pull request merged**